client/js/widgets/widget.js

```diff
@@ -1818,8 +1819,23 @@ export class Widget extends StateManaged {
   async move(coordGlobal, localAnchor) {
     const coordParent = this.coordParentFromCoordGlobal(coordGlobal);
     const offset = getOffset(this.coordParentFromCoordLocal(localAnchor), {x: this.get('x'), y: this.get('y')})
-    const newX = Math.round(coordParent.x + offset.x);
-    const newY = Math.round(coordParent.y + offset.y);
+    let newX = Math.round(coordParent.x + offset.x);
+    let newY = Math.round(coordParent.y + offset.y);
+    const limit = this.get('moveLimit');
+ 
+    if (limit.length) {
+      for (const p in limit[0]) {
```

Comment:
This for loop is completely unnecessary given that you're checking each property separately anyways. The way this works right now is for each property it checks if it's one of the 4 properties you care about.

Instead, since you know exactly which properties you care about you should check those directly, e.g. remove the for loop and do the following:

```js
if (limit[0].minX !== undefined) {
  newX = Math.max(limit[0].minX, newX);
}
...
```

I'm also not sure why limit is an array when you only use the first index. Instead, remove all of the [0] accessors and change limit to be an object, i.e. {} on line 45.

Reply:
> I'm also not sure why limit is an array when you only use the first index.

I'm not sure either.  Ask ChatGPT.

Thanks for the feedback.  I was following the model from grid, but I now remember that grid is an array of objects and I only need the one.

Reply:
> > I'm also not sure why limit is an array when you only use the first index.
> 
> I'm not sure either. Ask ChatGPT.

Fortunately I don't think it will be putting me out of a job just yet 😁